### PR TITLE
Add code coverage for ToolStripPanel_ToolStripPanelRowCollection

### PR DESCRIPTION
**Pull request by Zheng-Li01**
related https://github.com/dotnet/winforms/issues/10453

Proposed changes
Add unit tests for ToolStripPanel_ToolStripPanelRowCollection to test its properties & methods & events
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12111)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripPanel.ToolStripPanelRowCollectionTests.cs [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    private readonly ToolStripPanelRowCollection _toolStripPanelRowCollection;
+
+    public ToolStripPanel_ToolStripPanelRowCollectionTests()
+    {
+        _toolStripPanel = new();
+        _toolStripPanelRowCollection = new(_toolStripPanel);
+    }
+
+    public void Dispose() => _toolStripPanel.Dispose();
+
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripPanel.ToolStripPanelRowCollectionTests.cs:22]: Please add tests for 2 Constructors

```
 ToolStripPanelRowCollection toolStripPanelRowCollection = new(_toolStripPanel);
 ToolStripPanel toolStripPanel = toolStripPanelRowCollection.TestAccessor().Dynamic._owner;
 toolStripPanel.Should().BeSameAs(_toolStripPanel);
```


```
ToolStripPanelRow toolStripPanelRow1 = new(_toolStripPanel);
ToolStripPanelRow[] toolStripPanelRowArray= { toolStripPanelRow1 };
ToolStripPanelRowCollection toolStripPanelRowCollection = new(_toolStripPanel, toolStripPanelRowArray);
ToolStripPanel toolStripPanel = toolStripPanelRowCollection.TestAccessor().Dynamic._owner;

toolStripPanel.Should().BeSameAs(_toolStripPanel);
toolStripPanelRowCollection.Count.Should().Be(1);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripPanel.ToolStripPanelRowCollectionTests.cs [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    [WinFormsFact]
+    public void ToolStripPanelRowCollection_ConstructorWithOwner_SetsOwner()
+    {
+        ToolStripPanelRowCollection toolStripPanelRowCollection = new(_toolStripPanel);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripPanel.ToolStripPanelRowCollectionTests.cs:26]:
```suggestion
        using ToolStripPanelRowCollection toolStripPanelRowCollection = new(_toolStripPanel);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripPanel.ToolStripPanelRowCollectionTests.cs [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+        ToolStripPanel toolStripPanel = toolStripPanelRowCollection.TestAccessor().Dynamic._owner;
+        toolStripPanel.Should().BeSameAs(_toolStripPanel);
+    }
+
+    [WinFormsFact]
+    public void ToolStripPanelRowCollection_ConstructorWithOwnerAndRows_SetsOwnerAndAddsRows()
+    {
+        ToolStripPanelRow toolStripPanelRow1 = new(_toolStripPanel);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripPanel.ToolStripPanelRowCollectionTests.cs:34]:
```suggestion
        using ToolStripPanelRow toolStripPanelRow1 = new(_toolStripPanel);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripPanel.ToolStripPanelRowCollectionTests.cs [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    }
+
+    private ToolStripPanelRow AddRowToCollection(ToolStripPanelRowCollection collection, ToolStripPanel panel)
+    {
+        ToolStripPanelRow row = new(panel);
+        collection.Add(row);
+        return row;
+    }
+
+    private ToolStripPanelRow[] AddRowsToCollection(ToolStripPanelRowCollection collection, ToolStripPanel panel, int count)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripPanel.ToolStripPanelRowCollectionTests.cs:227]: Please add a method that enumerates an array of ToolStripPanelRows and disposes each of them.

```
private void DisposeRows(ToolStripPanelRow[] rows)
{
    int length = rows.Length;
    for (int i = 0; i < length; i++)
    {
        row[i].Dispose();
        row[i] = null;
    }
}
```

and call that to clean up tests that are using `AddRowsToCollection` in a try/finally statement. Of make `rows` a field in the test class for easy cleanup.

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

**Pull request merged**